crates/yttrium/src/chain_abstraction/api/route.rs

```diff
@@ -23,9 +23,21 @@ pub struct RouteRequest {
 #[serde(rename_all = "camelCase")]
 pub struct Metadata {
     pub funding_from: Vec<FundingMetadata>,
+    pub initial_transaction: InitialTransactionMetadata,
     pub check_in: u64,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "uniffi", derive(uniffi_macros::Record))]
+#[serde(rename_all = "camelCase")]
+pub struct InitialTransactionMetadata {
```

Comment:
Should this be some sort of enum saying ERC-20 vs swap or something? I suppose we could make a V2 route endpoint though

Reply:
Yeah let's add swap and be ready for other types in the future

Reply:
@geekbrother wdyt?

Reply:
I think we can always change the interface on the yttrium level. And not to think that future forward atm

Reply:
This interface also affects the Blockchain API response. But I agree we could change that easily too with new versions

Reply:
So let's keep as is then